micro-rdk/src/esp32/hcsr04.rs

```diff
@@ -86,7 +86,8 @@ struct IsrSharedState {
     //
     // 0: Starting state, ready to take a reading
     // i64 > 0: Millisecond timestamp of first edge of echo signal
-    timestamp: AtomicI64,
+    //TODO 32bit wide enough - To be fixed before merged
```

Comment:
I think you can probably just discard the high bits of the result from `esp_timer_get_time` to get a 32 bit value and use that, as long as you add a check that `when` > `prior` below. In the rare case that your low bits roll over between the issuing and receiving the pulse, resulting in `when` < `prior`, just don't notify, and the consumer will time out.

Reply:
:ok_hand: